geth/transactions/txqueue_manager.go

```diff
@@ -244,7 +250,7 @@ func (m *Manager) DiscardTransaction(id common.QueuedTxID) error {
 	}
 	err = m.txQueue.Done(id, gethcommon.Hash{}, queue.ErrQueuedTxDiscarded)
 	if m.notify {
-		NotifyOnReturn(tx)
+		NotifyOnReturn(tx, queue.ErrQueuedTxDiscarded)
```

Comment:
Similarly, maybe `ErrQueuedTxDiscarded` should be defined in `Manager`? `queue` also has no notion of discarding.